### ISSUE TITLE
feature request: silence the audio in skipping silence portion

### ISSUE THREAD

**Issue by lijovklm** [vantezzen/skip-silence#17]
hi,

It would be great to silence volume in the portion Which is skipped. Right now when the video is being skipped audio portion is also being rushed in the same speed as the video silence speed setting. Now if there is no background noise this works perfectly. However if there is background noise, the rushing of audio results in a strange sound. So would really appreciate if there’s a toggle button or mute that portion.

I believe CaryKH's tool had some kind of smoothening. I never felt this jerking sound with that.

Thank you.

**Comment by skoll43** [comment]
or maybe the option for just skip it?

**Comment by vantezzen** [comment]
> or maybe the option for just skip it?

Unfortunately, that is not possible due as the media is getting streamed from various websites and the extension can't analyze when a silent part end without getting to it.

**Comment by skoll43** [comment]
> > or maybe the option for just skip it?
> 
> Unfortunately, that is not possible as the media is getting streamed from various websites and the extension can't analyze when a silent part end without getting to it.

that would require something like https://github.com/TeamNewPipe/NewPipeExtractor for pre-streaming? (and maybe performance penalties with that)

**Comment by vantezzen** [comment]
Implemented this in the 3.2 update which I just uploaded to the Chrome Webstore and which should be published by them in the next few days.

> may worth to look for their implementation?

The problem is that NewPipe works in a very different environment: It has access to the whole video and can analyse it using native android features.
Skip Silence on the other hand, on a lot of websites (e.g. YouTube), can only analyse the currently playing audio and has no option to analyse future parts of the video. It can't download it up-front.
